### ISSUE TITLE
[Python] Implement new Concept Exercise: list-comprehensions

### ISSUE THREAD

**Issue by BethanyG** [exercism/v3#1361]
This issue describes how to implement the `list-comprehensions` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the syntax and variants of `list comprehensions` in Python. 

## Learning objectives

-  Understand how a `list comprehension` relates to the `loop` + `append` method of creating/computing a `List`.
-  Create a list comprehension from a `loop` + `append`
-  Create a list comprehension from a `string`,  `List`, or other `iterable` (_such as `zip()` or `dict.items()`_)
-  Use one or more conditions/operators/methods to filter comprehension inputs
-  Use methods or logic to format the elements (output members) of the comprehension
-  Create a _nested comprehension_ 
-  Create a _nested comprehension_ with one or more formatting or filtering conditions

## Out of scope

- Memory and performance characteristics and optimizations
- `generators` and `generator expressions` in `list comprehensions`
-  How `map()` and `filter()` relate to or differ from `list comprehensions`
-  Using the `assignment expression` (_walrus operator_) with a `list comprehension`

## Concepts

-  `list-comprehensions`

## Prerequisites

- [for-loops](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/for_loops.md)
- [builtin-types](https://github.com/exercism/v3/tree/master/languages/python/reference/concepts/builtin_types)
- [data-types](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/data_structures.md)
- [iterables](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iterables.md)
- [iteration](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iteration.md)
- [operators](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/operators.md)
- [comparison-operators](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/comparison_operators.md)
- [membership-testing](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/membership_testing.md)

## Resources to refer to

-  [List Comprehensions (Python official docs)](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)
- [Nested List Comprehensions (Python official docs)](https://docs.python.org/3/tutorial/datastructures.html#nested-list-comprehensions)
- [When to Use a List Comprehension in Python (Real Python)](https://realpython.com/list-comprehension-python/)
- [Python List Comprehensions:  Explained Visually (Trey Hunner)](https://treyhunner.com/2015/12/python-list-comprehensions-now-in-color/)

### Hints

- `List Comprehensions` section of the Python docs tutorial:  [List Comprehensions](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)
- This animated GIF from Trey Hunner:  [List Comprehensions: The Movie](https://treyhunner.com/images/list-comprehension-condition.gif)

### After

-  `comprehension syntax` for other data structures such as `sets` and `dictionaries`
- `generators` and `generator expressions`
- `generators` and `generator expressions` in `list comprehensions`

## Representer

No changes required.

## Analyzer

No changes rquired.

## Implementing

- Tests should be written using unittest.TestCase, and the test file named list_comprehensions_test.py.
- [How to Implement a Concept Exercise in Python](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md)
- [make-concept-exercise Utility](https://github.com/exercism/v3/tree/master/languages/python/bin)

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
